### DISCUSSION TITLE
Works locally, but on Github Actions: build can't transfer own Github hosted parent pom, receives Bad Request and 'parent.relativePath' points at wrong local POM

### DISCUSSION THREAD

**Discussion by joergi** [community/community#26924] [Actions]
<p>I’m at the moment building a project.<br>
This project should use my <a href="https://github.com/joergi/parent-jdk11-mongo/" rel="nofollow noopener">parent.pom</a> from the <a href="https://github.com/joergi/parent-jdk11-mongo/packages/228373" rel="nofollow noopener">Github registry</a>.</p>
<p>It works perfectly on my machine, even if I delete the parent-pom from my .m2 repository<br>
Unfortunately, I always get an <a href="https://github.com/joergi/basics/runs/1072139522#step:4:13" rel="nofollow noopener">error when Github tries to run the tests</a> after each push, because the parent-pom can’t be resolved.</p>
<pre><code class="lang-auto">    [ERROR] Some problems were encountered while processing the POMs:
    [FATAL] Non-resolvable parent POM for io.joergi:basics:0.0.1-SNAPSHOT: Could not 
    transfer artifact io.joergi:parent-jdk11-mongo:pom:2.3.0-1.RELEASE from/to github 
    (https://maven.pkg.github.com/joergi/parent-jdk11-mongo): Transfer failed for 
    https://maven.pkg.github.com/joergi/parent-jdk11-mongo/io/joergi/parent-jdk11-
    mongo/2.3.0-1.RELEASE/parent-jdk11-mongo-2.3.0-1.RELEASE.pom 400 Bad Request and 
    'parent.relativePath' points at wrong local POM @ line 5, column 10
</code></pre>
<p>The parent-pom has this definition:</p>
<pre><code class="lang-auto">  &lt;groupId&gt;io.joergi&lt;/groupId&gt;
	&lt;artifactId&gt;parent-jdk11-mongo&lt;/artifactId&gt;
	&lt;version&gt;2.3.0-1.RELEASE&lt;/version&gt;
	&lt;packaging&gt;pom&lt;/packaging&gt;
	&lt;name&gt;joergi.io parent-jdk11-mongo&lt;/name&gt;

	&lt;parent&gt;
		&lt;groupId&gt;org.springframework.boot&lt;/groupId&gt;
		&lt;artifactId&gt;spring-boot-starter-parent&lt;/artifactId&gt;
		&lt;version&gt;2.3.3.RELEASE&lt;/version&gt;
		&lt;relativePath /&gt; &lt;!-- lookup parent from repository --&gt;
	&lt;/parent&gt;
</code></pre>
<p>My <a href="https://github.com/joergi/basics" rel="nofollow noopener">new project</a> use Github actions for automatic testing.<br>
It integrates the parent pom like this:</p>
<pre><code class="lang-auto">
    &lt;parent&gt;
		&lt;groupId&gt;io.joergi&lt;/groupId&gt;
		&lt;artifactId&gt;parent-jdk11-mongo&lt;/artifactId&gt;
		&lt;version&gt;2.3.0-1.RELEASE&lt;/version&gt;
	&lt;/parent&gt;

	&lt;distributionManagement&gt;
		&lt;repository&gt;
			&lt;id&gt;github&lt;/id&gt;
			&lt;name&gt;GitHub joergi Apache Maven Packages&lt;/name&gt;
			&lt;url&gt;https://maven.pkg.github.com/joergi/parent-jdk11-mongo&lt;/url&gt;
		&lt;/repository&gt;
	&lt;/distributionManagement&gt;
</code></pre>
<p>I also have a <a href="https://github.com/joergi/basics/blob/b3adc829e2712d9a498910c8ecc0ab9d7ea2e091/settings.xml" rel="nofollow noopener">settings.xml</a> in the new project</p>
<pre><code class="lang-auto">  &lt;activeProfiles&gt;
        &lt;activeProfile&gt;github&lt;/activeProfile&gt;
      &lt;/activeProfiles&gt;
    
      &lt;profiles&gt;
        &lt;profile&gt;
          &lt;id&gt;github&lt;/id&gt;
          &lt;repositories&gt;
            &lt;repository&gt;
              &lt;id&gt;github&lt;/id&gt;
              &lt;name&gt;GitHub joergi Apache Maven Packages&lt;/name&gt;
              &lt;url&gt;https://maven.pkg.github.com/joergi/parent-jdk11-mongo&lt;/url&gt;
            &lt;/repository&gt;
          &lt;/repositories&gt;
        &lt;/profile&gt;
      &lt;/profiles&gt;
    
      &lt;servers&gt;
        &lt;server&gt;
          &lt;id&gt;github&lt;/id&gt;
          &lt;username&gt;joergi&lt;/username&gt;
          &lt;password&gt;${{ secrets.PACKAGES_READ_ONLY }}&lt;/password&gt;
        &lt;/server&gt;
      &lt;/servers&gt;
</code></pre>
<p>The secret is in my Github secrets. If I remove or change it, I’m not authorized anymore, so this works.</p>
<p>Can one of you give me a hint, what is wrong</p>
<p>(I already posted it on <a href="https://stackoverflow.com/posts/63743022" rel="nofollow noopener">StackOverflow</a>, but no answer yet)</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/joergi">@joergi</a> ,</p>
<p>I have created an issue ticket (<a href="https://github.com/actions/virtual-environments/issues/1554" rel="nofollow noopener"><strong>actions/virtual-environments#1554</strong></a>) to help you report the problem the appropriate engineering team for further investigation and evaluation.</p>
<p>You can follow this issue ticket and add your comment to it.</p>
<br>
<p>In addition, you mentioned,</p>
<blockquote>
<p>It works perfectly on my machine, even if I delete the parent-pom from my .m2 repository</p>
</blockquote>
<p>You can try to install a self-hosted runner on your local machine to run the workflow to see if the error still occurs.</p>

**Comment by joergi** [comment]
<p>how do I start / use a self-hosted runner?<br>
never tried it</p>

**Comment by joergi** [comment]
<p>Honestly, at the moment I’m not sure if the setting.xml uses the github secret.<br>
because if I look into my Github tokens, I can see that it’s never used…</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/joergi">@joergi</a> ,</p>
<blockquote>
<p>how do I start / use a self-hosted runner?</p>
</blockquote>
<p>You can see “<a href="https://docs.github.com/en/actions/hosting-your-own-runners/adding-self-hosted-runners" rel="nofollow noopener"><strong>Adding self-hosted runners</strong></a>” to get more details about adding self-hosted runners.<br>
And more information about self-hosted runners, you can see “<a href="https://docs.github.com/en/actions/hosting-your-own-runners" rel="nofollow noopener"><strong>Hosting your own runners</strong></a>”.</p>
<br>
<blockquote>
<p>Honestly, at the moment I’m not sure if the setting.xml uses the github secret.<br>
because if I look into my Github tokens, I can see that it’s never used…</p>
</blockquote>
<p>To use the <strong>GITHUB_TOKEN</strong> or a personal access token (PAT) in the <strong>setting.xml</strong>, you can try like as the following:</p>
<ul>
<li>
<p>In the <strong>setting.xml</strong> file, you can set like as this.</p>
<pre><code class="lang-xml">&lt;settings . . .&gt;
  . . .
  &lt;servers&gt;
    &lt;server&gt;
      &lt;id&gt;github&lt;/id&gt;
      &lt;username&gt;USERNAME&lt;/username&gt;
      &lt;password&gt;${MVN_AUTH_TOKEN}&lt;/password&gt;
    &lt;/server&gt;
  &lt;/servers&gt;
&lt;/settings&gt;
</code></pre>
</li>
<li>
<p>In the workflow file, you need to map the <strong>GITHUB_TOKEN</strong> or the PAT as an environment variable.<br>
For example:</p>
<pre><code class="lang-yaml">- name: Publish to GitHub Packages
  run: mvn deploy -s settings.xml
  env:
    MVN_AUTH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
</code></pre>
</li>
</ul>

**Comment by joergi** [comment]
<p>Thanks a lot.<br>
This works.<br>
you can close the other ticket, I guess!?</p>
<p>(no I have some other error, but I know how to solve it)</p>
